instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/internal/HttpConstants.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.api.internal;
+
+import static java.util.Arrays.asList;
+import static java.util.Collections.unmodifiableSet;
+
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * This class is internal and is hence not for public use. Its APIs are unstable and can change at
+ * any time.
+ */
+public final class HttpConstants {
+
+  public static final Set<String> KNOWN_METHODS =
+      unmodifiableSet(
+          new HashSet<>(
+              asList(
+                  "CONNECT", "DELETE", "GET", "HEAD", "OPTIONS", "PATCH", "POST", "PUT", "TRACE")));
+
+  public static final String _OTHER = "_OTHER";
```

Comment:
why the `_`?

Reply:
It was chosen because it makes the "other" default value look like it's a kinda special thing (which it is)

> If the HTTP request method is not known to instrumentation, it MUST set the `http.request.method` attribute to `_OTHER` and, except if reporting a metric, MUST set the exact method received in the request line as value of the `http.request.method_original` attribute.

Reply:
If it comes from the spec then so it must be. In my opinion using `_OTHER` is a mistake, there is no way that this is not going to confuse users. Everybody who sees it will assume that this is bug/typo.

Reply:
🤷 
I don't really have a strong opinion on the value itself; I do agree that you need to have some sort of indication that it's some other unrecognized value though.

Reply:
this is probably the most relevant comment why `_OTHER` was chosen: https://github.com/open-telemetry/semantic-conventions/pull/17#discussion_r1230390256

I could see backends potentially displaying this metric value with a special tooltip to help ease user confusion